packages/twenty-front/src/modules/object-record/record-field/hooks/usePersistField.ts

```diff
@@ -31,15 +36,46 @@ import { isFieldRelationValue } from '../types/guards/isFieldRelationValue';
 import { isFieldText } from '../types/guards/isFieldText';
 import { isFieldTextValue } from '../types/guards/isFieldTextValue';
 
-export const usePersistField = () => {
-  const {
-    entityId,
-    fieldDefinition,
-    useUpdateRecord = () => [],
-  } = useContext(FieldContext);
+const useUpsertRecord = () => {
+  const { pendingRecordId } = usePendingRecordId();
 
+  const { useUpdateRecord = () => [] } = useContext(FieldContext);
   const [updateRecord] = useUpdateRecord();
 
+  const objectNamePlural = useParams().objectNamePlural ?? '';
```

Comment:
is it safe to call this here ? or should usePersistField remain agnostic to current page params?

Reply:
discussed with you, we can leverage fieldContext!